src/main/java/org/broadinstitute/hellbender/tools/funcotator/FuncotatorDataSourceRetriever.java

```diff
@@ -0,0 +1,221 @@
+
+        // Set up and initiate our download:
+        final NioFileCopier xerox = NioFileCopier.create(dataSourcesPath, outputDestination, overwriteOutputFile);
+        xerox.initiateCopy();
+
+        return xerox;
+    }
+
+    private void validateIntegrity(final NioFileCopier fileCopier, final Path remoteSha256Path) {
+
+        // Get the SHA 256 file:
+        final Path localSha256Path = getSha256File(remoteSha256Path);
+
+        // Read the sha256sum into memory:
+        final String expectedSha256Sum = readSha256SumFromFile(localSha256Path);
+
+        // Calculate the sha256sum of the data sources file:
+        final boolean downloadedDataSourcesAreValid = fileCopier.validateIntegrity(expectedSha256Sum, DigestUtils::sha256Hex);
+
+        // verify the hashes are the same:
+        if ( !downloadedDataSourcesAreValid ) {
+            throw new GATKException("ERROR: downloaded data sources are corrupt!  Unexpected checksum: " + fileCopier.getLatestChecksum() + " != " + expectedSha256Sum);
+        }
+        else {
+            logger.info("Data sources are valid.");
+        }
+    }
+
+    private Path getSha256File(final Path remoteSha256Path) {
+        final Path localSha256Path = getDataSourceLocalPath(remoteSha256Path);
+        logger.info("Retrieving expected checksum file...");
+        NioFileCopier.create(remoteSha256Path, localSha256Path, overwriteOutputFile).setSilentCopy(true).initiateCopy();
+        logger.info("File transfer complete!");
+
+        return localSha256Path;
+    }
+
+    private String readSha256SumFromFile(final Path localSha256Path) {
+        final String expectedSha256Sum;
+        try {
+            logger.info("Collecting expected checksum...");
+            expectedSha256Sum = FileUtils.readFileToString(localSha256Path.toFile(), Charset.defaultCharset());
+            logger.info("Collection complete!");
+        }
+        catch ( final IOException ex ) {
+            throw new GATKException("Could not read in sha256sum from file: " + localSha256Path.toUri().toString(), ex);
+        }
+
+        // Clean up and return the checksum:
+        return cleanExpectedSha256SumString(expectedSha256Sum);
+    }
+
+    private Path getDataSourceLocalPath(final Path dataSourcesPath) {
+        return IOUtils.getPath(dataSourcesPath.getFileName().toString());
+    }
+
+    private String cleanExpectedSha256SumString(final String expectedSha256SumString) {
+
+        String cleanString = expectedSha256SumString.trim().toLowerCase();
+
+        // The format of the file can contain the checksum, followed by the file name.
+        // If this is the case, we need to truncate the string:
+        while ( cleanString.contains(" ") ) {
+            cleanString = cleanString.substring(0, cleanString.indexOf(" "));
+        }
+        while (  cleanString.contains("\t") ) {
+            cleanString = cleanString.substring(0, cleanString.indexOf("\t"));
+        }
```

Comment:
Can't you just call `String.trim()` here on `expectedSha256SumString` to get rid of leading/trailing whitespace, and eliminate this method?

Reply:
No - the format of the files can either be a raw checksum, or the following:

`<CHECKSUM><WHITESPACE><FILENAME>`

So I need to make sure only the checksum is used.  I can clean this up a little though.